juju/juju
# 2.3 into develop

## Description of change

Lots of patches in 2.3 had not been forward ported to develop. Rather than split them out individually, we again just merge 2.3 directly into develop, and resolve a few of the conflicts that it generated.

## QA steps

None, CI should pass on 2.3 and develop, and should continue to do so after this change. Some things like "juju controller-config" should be available in develop after this change.

## Documentation changes

None for this patch.

## Bug reference

See the individual changes that were brought in (~72 total commits representing 14 merges).
 * Merge pull request #8323 from wupeka/2.3-networking-issues
 * Merge pull request #8332 from nskaggs/support-lxd-snap-socket
 * Merge pull request #8343 from babbageclunk/audit-log-fixups
 * Merge pull request #8342 from babbageclunk/audit-config-update
 * Merge pull request #8340 from anastasiamac/auto-upgrade-lp1745951
 * Merge pull request #8331 from AdrianGPrado/1744469_upgrade-juju--dry-run
          LP [#1744469](https://bugs.launchpad.net/juju/+bug/1744469)
 * Merge pull request #8315 from babbageclunk/controller-config-update
          https://bugs.launchpad.net/juju/+bug/1745231
 * Merge pull request #8324 from anastasiamac/dump-logs-when-no-model
 * Merge pull request #8316 from anastasiamac/model-dump-not-found-23
 * Merge pull request #8314 from anastasiamac/deal-not-found-23
          https://bugs.launchpad.net/juju/+bug/1745231
 * Merge pull request #8303 from jameinel/2.2-into-develop
 * Merge pull request #8281 from wupeka/2.3-fix-race-TestLoginAddsAuditConversationEventually
 * Merge pull request #8293 from Veebers/ci-bring-2.3-uptodate
